packages/ui/src/pages/Beans/BeansPage.tsx

```diff
@@ -0,0 +1,31 @@
+import { Title } from '@patternfly/react-core';
+import {FunctionComponent, useContext} from 'react';
+import {MetadataEditor} from "../../components/MetadataEditor";
+import { useSchemasStore } from '../../store';
+import { EntitiesContext } from '../../providers/entities.provider';
+import {BeansDeserializer} from "@kaoto-next/camel-catalog/types";
+
+export const BeansPage: FunctionComponent = () => {
+  const schemaMap = useSchemasStore((state) => state.schemas);
+  const entitiesContext = useContext(EntitiesContext);
+  const metadata = entitiesContext?.entities ?? [];
+  const beansData =
+    (metadata.find(item => item.type === 'beans') ?? {}) as {beans: BeansDeserializer};
+
+  function onChangeModel(model: BeansDeserializer) {
+    beansData.beans = [ ...model ];
+    entitiesContext?.updateCodeFromEntities();
+  }
+
+  return (
+    <>
+      <Title headingLevel="h1">Beans Configuration</Title>
+      <MetadataEditor
+        name="Beans"
+        schema={schemaMap.beans.schema}
```

Comment:
@lordrip `schemaMap.beans` here is sometimes `undefined`, usually if I reload the browser once or twice then it loads as expected, is there a something wrong on how schema store is used?

Reply:
mmm, I think this might be related to the initial problem, probably fetching the schemas is not a blocking operation so the beans UI gets rendered before having the schema

Reply:
OK so leave it as-is, hoping that's better for prod build, or we'll revisit separately